### ISSUE TITLE
A verify_sources Fault cannot be cleared by passing, so it outlives the fix and withdraws a layer anyway

### ISSUE THREAD

**Issue by cgbarlow** [cgbarlow/braintrust#283]
## What happens

Every Fault report braintrust writes ends with the same promise:

> braintrust clears it when the assertion passes, not when this issue is closed.

For `verify_sources` that sentence cannot be kept. `verifySources` calls `openFault` when a check fails and has **no path to `clearFault` at all** — passing does nothing. And nothing in the scheduled job re-runs the check either; `verifySources` is reached from exactly one place, the MCP tool handler. So the fault can only ever open, and only a client happening to call the tool can even re-observe it.

The consequence is that a `verify_sources` Fault outlives any fix to the detector that raised it. It keeps re-printing the evidence it was opened on — which reads as current, because nothing marks it historical — and it escalates on schedule regardless.

## Measured

2026-08-15. #280 fixed the false positives that had `verify_sources` convicting braintrust's own record handover of forgery. After deploying it, the re-run #277 asks for reported:

```
#277 re-run — 5 serving Persona(s)
  chris-barlow:        4/4 sourced, 0 unsourced, 0 never claimed
  ethan-mollick:       4/4 sourced, 0 unsourced, 0 never claimed
  matt-pocock:         4/4 sourced, 0 unsourced, 0 never claimed
  nate-b-jones:        4/4 sourced, 0 unsourced, 0 never claimed
  stuart-winter-tear:  4/4 sourced, 0 unsourced, 0 never claimed
  TOTAL: 20/20 sourced, 0 unsourced, 0 never claimed
```

All three open faults survived that unchanged, and were still scheduled to escalate the next morning — withdrawing `reasoning` from `ethan-mollick`, `matt-pocock` and `stuart-winter-tear` on evidence that had just been disproved twenty times over.

The scheduled run before it showed the same thing from the other side: the fault rendered with #280's new guarantee text wrapped around a `verified against: braintrust_items.body_text — indexOf` line, which is the pre-#280 constant. New code cannot emit that string. The report was a stale observation in a fresh frame, and nothing about it said so.

## Why it matters beyond this one fault

#276 exists so that a shipped fix is verified within a day rather than at the weekly sweep. That works for Interrogation assertions, which are re-asked and cleared by passing — it visibly worked the same night, restoring `reasoning` across the fleet. It does not work for any tool-driven fault, because there is nothing to re-ask and no way to pass.

#280 also gave `verify_sources` the power to withdraw `reasoning`, which it did not have before. So a fault that cannot clear itself now has teeth.

## What to build

- A `verify_sources` check where every sentence passes clears the fault for that person, exactly as an Interrogation assertion passing does.
- Clearing happens on a pass, never merely because the tool was called — a failing reply must leave the fault standing.
- The scheduled run has some way to re-observe a tool-driven fault, so clearing does not depend on a client happening to call the tool. The probe from #277 is the obvious candidate for what that runs.
- A Fault report distinguishes evidence gathered now from evidence stored when the fault opened, so a stale observation cannot be read as a current one.

## Acceptance criteria

- [ ] A verification in which every sentence checks out clears that person's `verify_sources` fault
- [ ] A verification with any failure leaves the fault open and does not clear it
- [ ] Clearing is scoped to that assertion and that person; no other fault is touched
- [ ] A fault that has been cleared stops appearing in run output and stops escalating
- [ ] The scheduled run can re-observe a tool-driven fault without a client calling the tool
- [ ] A re-printed Fault report says when its evidence was gathered

## Blocked by

- None — can start immediately.
